pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/loader/SegmentPreProcessor.java

```diff
@@ -104,77 +105,11 @@ public void process()
         LOGGER.warn("Skip creating default columns for segment: {} without schema", _segmentMetadata.getName());
       }
 
-      // Create column inverted indices according to the index config.
-      InvertedIndexHandler invertedIndexHandler =
-          new InvertedIndexHandler(_indexDir, _segmentMetadata, _indexLoadingConfig, segmentWriter);
-      invertedIndexHandler.createInvertedIndices();
-
-      // Create column range indices according to the index config.
-      RangeIndexHandler rangeIndexHandler =
-          new RangeIndexHandler(_indexDir, _segmentMetadata, _indexLoadingConfig, segmentWriter);
-      rangeIndexHandler.createRangeIndices();
-
-      // Create text indices according to the index config.
-      Set<String> textIndexColumns = _indexLoadingConfig.getTextIndexColumns();
-      if (!textIndexColumns.isEmpty()) {
-        TextIndexHandler textIndexHandler =
-            new TextIndexHandler(_indexDir, _segmentMetadata, textIndexColumns, segmentWriter);
-        textIndexHandler.createTextIndexesOnSegmentLoad();
-      }
-
-      Set<String> fstIndexColumns = _indexLoadingConfig.getFSTIndexColumns();
-      if (!fstIndexColumns.isEmpty()) {
-        LuceneFSTIndexHandler luceneFSTIndexHandler =
-            new LuceneFSTIndexHandler(_indexDir, _segmentMetadata, fstIndexColumns, segmentWriter);
-        luceneFSTIndexHandler.createFSTIndexesOnSegmentLoad();
-      }
+      createMissingIndices(segmentWriter);
```

Comment:
I somehow feel this refactor is not improving the readability.
We can make star-tree handling a separate method because the logic is more complicated.

Reply:
I kinda feel it natural to group them into one sub method, to make the main flow clear here. no strong preference tho.